### DISCUSSION TITLE
TechEmPower Benchmarks

### DISCUSSION THREAD

**Discussion by cirospaciari** [emmett-framework/granian#62] [Q&A] [answered]
Can I add basic plaintext/JSON benchmarks for Falcon, FastAPI, Emmett, Django and Flask powered by granian?

Mostly of benchmarks are using meinheld for WSGI and uvicorn+gunicorn, I created for socketify with the tag [socketify WSGI] and [socketify ASGI] I can do the same using [granian ASGI] and [granian WSGI] with PyPy variants too.

**Comment by gi0baro** [comment, marked as answer]
@cirospaciari that sound a good "marketing strategy", but I'm not a big fan of wasting computing resources.
IMHO "platforms" like `granian` should just be tested in isolation, as it is comparable only with other servers, not frameworks.

For instance: if RPS of uvicorn vs granian produce a ∂R, then whatever framework you might test will just have the same ∂R comparing it running with uvicorn or granian.
I would prefer to see some "rationalisation" on TechEmpower banchmarks, starting with having a clean gunicorn implementation, so we can test ASGI and WSGI in "plain mode". Then we can just demonstrate the performance of the server implementation, while the final RPS of the framework will be just a ~constant reduced ratio from the server the framework uses.

In addition: no need to add granian benchmarks for Emmett, as in the next version it will just use Granian in RSGI.
